src/bin/wasmer.rs

```diff
@@ -295,60 +297,81 @@ fn execute_wasm(options: &Run) -> Result<(), String> {
         let mut emscripten_globals = wasmer_emscripten::EmscriptenGlobals::new(&module);
-        (
-            InstanceABI::Emscripten,
-            wasmer_emscripten::generate_emscripten_env(&mut emscripten_globals),
-            Some(emscripten_globals), // TODO Em Globals is here to extend, lifetime, find better solution
+        let import_object = wasmer_emscripten::generate_emscripten_env(&mut emscripten_globals);
+        let mut instance = module
+            .instantiate(&import_object)
+            .map_err(|e| format!("Can't instantiate module: {:?}", e))?;
+
+        wasmer_emscripten::run_emscripten_instance(
+            &module,
+            &mut instance,
+            if let Some(cn) = &options.command_name {
+                cn
+            } else {
+                options.path.to_str().unwrap()
+            },
+            options.args.iter().map(|arg| arg.as_str()).collect(),
         )
+        .map_err(|e| format!("{:?}", e))?;
     } else {
         if cfg!(feature = "wasi") && wasmer_wasi::is_wasi_module(&module) {
-            (
-                InstanceABI::WASI,
-                wasmer_wasi::generate_import_object(
-                    if let Some(cn) = &options.command_name {
-                        [cn.clone()]
-                    } else {
-                        [options.path.to_str().unwrap().to_owned()]
-                    }
-                    .iter()
-                    .chain(options.args.iter())
-                    .cloned()
-                    .map(|arg| arg.into_bytes())
+            let import_object = wasmer_wasi::generate_import_object(
+                if let Some(cn) = &options.command_name {
+                    [cn.clone()]
+                } else {
+                    [options.path.to_str().unwrap().to_owned()]
+                }
+                .iter()
+                .chain(options.args.iter())
+                .cloned()
+                .map(|arg| arg.into_bytes())
+                .collect(),
+                env::vars()
+                    .map(|(k, v)| format!("{}={}", k, v).into_bytes())
                     .collect(),
-                    env::vars()
-                        .map(|(k, v)| format!("{}={}", k, v).into_bytes())
-                        .collect(),
-                    options.pre_opened_directories.clone(),
-                ),
-                None,
-            )
-        } else {
-            (
-                InstanceABI::None,
-                wasmer_runtime_core::import::ImportObject::new(),
-                None,
-            )
-        }
-    };
+                options.pre_opened_directories.clone(),
+            );
 
-    let mut instance = module
-        .instantiate(&import_object)
-        .map_err(|e| format!("Can't instantiate module: {:?}", e))?;
+            let instance = module
+                .instantiate(&import_object)
+                .map_err(|e| format!("Can't instantiate module: {:?}", e))?;
 
-    webassembly::run_instance(
```

Comment:
Maybe this method could be deprecated if it will no longer be called.

Reply:
Yeah, this is a good idea.